### PR TITLE
[WIP] drivers: imx: Dummy DMA driver

### PR DESCRIPTION
**Pull request by paulstelian97**
For the communication between host and DSP a dummy DMA driver which
simply does memcpy is sufficient (given that a sizeable amount of
physical memory is shared)

This commit holds an attempt at the dummy DMA driver, as well as
additional host support (support for host page tables and the proper
usage of the dummy DMA driver for transfering the page tables).

I attempted to also add support for the DMA trace, however it is not
currently working.

Signed-off-by: Paul Olaru <paul.olaru@nxp.com>

Draft due to lack of good testing and due to DMA trace not working (or at least not seeming to work).

See also: #1804, #1806 (related, currently WIP pull requests for this platform)

### CODE REVIEW

File: src/platform/imx8/platform.c [hunk 1]
```diff
@@ -124,14 +124,14 @@ struct timesource_data platform_generic_queue[] = {
...
-
 int platform_boot_complete(uint32_t boot_message)
 {
 	mailbox_dspbox_write(0, &ready, sizeof(ready));
 	mailbox_dspbox_write(sizeof(ready), &sram_window,
 			     sram_window.ext_hdr.hdr.size);
 
 	/* now interrupt host to tell it we are done booting */
+
 	imx_mu_xcr_rmw(IMX_MU_xCR_GIRn(1), 0);
```
Comment (dbaluta) [inline src/platform/imx8/platform.c]: remove new line here

**Comment by dbaluta** [comment]
we can leave DMA trace support out for now. Also, please please add more details in the commit message about how dummy DMA works.

**Comment by paulstelian97** [comment]
> we can leave DMA trace support out for now. Also, please please add more details in the commit message about how dummy DMA works.

I have separated DMA trace support for now into a separate commit (it may not be working due to the Dummy DMA driver itself rather than the specific DMA trace changes).

Added the additional explanation, is it alright?

**Comment by dbaluta** [comment]
@Paul, general comment. I think it would be better to do the review for now internalu on your WIP branch for ESAI, EDMA, dummy DMA. So please create the PR on your branch so we can make the review there and dont waste reviewrs time until we have something that it is clean and working. Sorry for misguiding you.

**Comment by paulstelian97** [comment]
> @ paul, general comment. I think it would be better to do the review for now internalu on your WIP branch for ESAI, EDMA, dummy DMA. So please create the PR on your branch so we can make the review there and dont waste reviewrs time until we have something that it is clean and working. Sorry for misguiding you.

Will do.

**Comment by paulstelian97** [comment]
Replaced by #1813

**Pull request closed without merging**